Task 4/Task 4/Task 4/FileWatcher.cs

```diff
@@ -6,20 +6,27 @@ namespace Task_4
 {
     public class FileWatcher
     {
+        public enum FileActions
+        {
+            Default = 0,
+            Create = 1,
+            Delete = 2,
+            Rename = 3,
+            Change = 4
+        }
+
         private List<FileEventsInfo> _log = new List<FileEventsInfo> { };
 
-        public void WatchFolder (string folderPath, string logPath)
+        public void WatchFolder (string folderPath, LogsSerializer log)
         {
-            if (folderPath == null || logPath == null || folderPath == String.Empty || logPath == String.Empty)
+            if (folderPath == null ||folderPath == String.Empty)
                 throw new IOException("Path to file or folder can't be null or empty");
```

Comment:
That's an argument exception, not IOException - a validation error, not an error related to input-output failures